webbeans-junit5/src/main/java/org/apache/openwebbeans/junit5/internal/CdiExtension.java

```diff
@@ -172,4 +182,76 @@ private void doClose(final SeContainer container)
             }
         });
     }
+
+    private Cdi getCdiConfig(ExtensionContext extensionContext)
+    {
+        return AnnotationUtils.findAnnotation(extensionContext.getElement(), Cdi.class).orElse(null);
+    }
+
+    private SeContainer getContainer()
+    {
+        if (testInstanceContainer != null)
+        {
+            return testInstanceContainer;
+        }
+        else
+        {
+            return reusableContainer;
+        }
+    }
+
+    @Override
+    public boolean supportsParameter(ParameterContext parameterContext, ExtensionContext extensionContext)
+            throws ParameterResolutionException
+    {
+        final Cdi config = extensionContext.getParent().map(this::getCdiConfig).orElse(null);
```

Comment:
`extensionContext` here is a MethodExtensionContext while `extensionContext` in beforeAll is a ClassExtensionContext which is the parent. Methods always have classes as parents and there also will always be a parent class for a method.
However, I'm not fully satisfied that this will work with all situations such as dynamic tests, nested test classes, or whatever else. For a beginning I tend to opt that this is no worse than before but it can be further elaborated as desired.

Reply:
Normally jupiter goes through the hierarchy when requesting an annotation so we shouldn't need to do getparent anytime but just use a store in the extensioncontext, no?